beets/importer.py

```diff
@@ -675,10 +688,16 @@ def find_duplicates(self, lib):
 
         duplicates = []
         task_paths = {i.path for i in self.items if i}
-        duplicate_query = dbcore.AndQuery((
-            dbcore.MatchQuery('albumartist', artist),
-            dbcore.MatchQuery('album', album),
-        ))
+        keys = config['import']['duplicate_keys'].as_str().split()
+        info = self.chosen_info().copy()
+        info['albumartist'] = artist
+        album = library.Album(None, **info)
+        subqueries = []
+        for key in keys:
+            value = album.get(key)
+            fast = key in library.Album.item_keys
```

Comment:
This should probably use the *album-level* keys, not the item-level keys (`item_keys`), because we're doing an album-level query.

It's possible it would be a good idea to introduce some kind of utility method on `Album` and `Item` just for constructing `MatchQuery` objects that correctly set the `fast` flag…